pkg/api/persistentvolume/util.go

```diff
@@ -28,6 +28,7 @@ func DropDisabledFields(pvSpec *api.PersistentVolumeSpec, oldPVSpec *api.Persist
 	if !utilfeature.DefaultFeatureGate.Enabled(features.ExpandCSIVolumes) && !hasExpansionSecrets(oldPVSpec) {
```

Comment:
With this code, if there is ControllerExpandSecretRef  the old object and does not have `NodeExpandSecretRef`, the code does not clear `NodeExpandSecretRef` in the new object. Which is IMO wrong.

I think you should completely split `ControllerExpandSecretRef` and `NodeExpandSecretRef` handling, i.e. have `hasControllerExpansionSecrets()` and `hasNodeExpansionSecrets()` as separate functions and separate `if` here.

Reply:
good suggestion